src/SelfRoleManager.ts

```diff
@@ -255,10 +259,15 @@ export class SelfRoleManager extends EventEmitter {
     const userWantsToRemoveRole = isButtonInteraction ? memberHasRole : memberHasRole && ((!rteData.removeOnReact && isReactionRemoval) || (rteData.removeOnReact && !isReactionRemoval));
     const userWantsToAddRole = isButtonInteraction ? !memberHasRole : !memberHasRole && ((!rteData.removeOnReact && !isReactionRemoval) || (rteData.removeOnReact && isReactionRemoval));
     const role: Role = rteData.role instanceof Role ? rteData.role : await userAction.message.guild.roles.fetch(rteData.role);
+    const userNotMetDependencies = rteData.dependencies?.some((role: RoleResolvable) => memberRoles.find((memberRole: Role) => (role instanceof Role ? role === memberRole : role === memberRole.id)) === undefined) ?? false;
+
     let updatedMember: GuildMember;
     switch (true) {
       case userWantsToAddRole && maxRolesReached:
-        this.emit(SelfRoleManagerEvents.maxRolesReach, member, userAction, memberRoles.length, channelOptions.maxRolesAssigned);
+        this.emit(SelfRoleManagerEvents.maxRolesReach, member, userAction, memberManagedRoles.length, channelOptions.maxRolesAssigned);
+        break;
+      case userWantsToAddRole && userNotMetDependencies:
```

Comment:
Following the previous comment, don't forget that you should probably inverse the boolean here.
```suggestion
      case userWantsToAddRole && !userMeetsDependencies :
```